_ml-commons-plugin/extensibility/bedrock-connector.md

```diff
@@ -0,0 +1,274 @@
+ "status": "CREATED"
+}
+```
+
+To learn more about model groups, see [Model access control]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control/).
+
+## Step 2: Create a Bedrock connector
+
+Create a standalone Bedrock connector by sending the following request:
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "Bedrock multimodal embedding model",
+  "description": "Bedrock multimodal model which supports both text and image embeddings",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "parameters": {
+    "region": "us-east-1",
+    "service_name": "bedrock"
+  },
+  "credential": {
+    "access_key": "<REPLACE WITH YOUR BEDROCK ACCESS KEY>", 
+    "secret_key": "<REPLACE WITH YOUR BEDROCK SECRET KEY>"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://bedrock.us-east-1.amazonaws.com/model/amazon.titan-e1m-medium/invoke",
+      "headers": {
+        "content-type": "application/json",
+        "x-amz-content-sha256": "required"
+      },
+      "request_body": "{ \"inputText\": \"${parameters.inputText:-null}\", \"inputImage\": \"${parameters.inputImage:-null}\" }",
+      "pre_process_function": "\n    StringBuilder parametersBuilder = new StringBuilder(\"{\");\n    if (params.text_docs.length > 0 && params.text_docs[0] != null) {\n      parametersBuilder.append(\"\\\"inputText\\\":\");\n      parametersBuilder.append(\"\\\"\");\n      parametersBuilder.append(params.text_docs[0]);\n      parametersBuilder.append(\"\\\"\");\n      \n      if (params.text_docs.length > 1 && params.text_docs[1] != null) {\n        parametersBuilder.append(\",\");\n      }\n    }\n    \n    \n    if (params.text_docs.length > 1 && params.text_docs[1] != null) {\n      parametersBuilder.append(\"\\\"inputImage\\\":\");\n      parametersBuilder.append(\"\\\"\");\n      parametersBuilder.append(params.text_docs[1]);\n      parametersBuilder.append(\"\\\"\");\n    }\n    parametersBuilder.append(\"}\");\n    \n    return  \"{\" +\"\\\"parameters\\\":\" + parametersBuilder + \"}\";",
+      "post_process_function": "\n      def name = \"sentence_embedding\";\n      def dataType = \"FLOAT32\";\n      if (params.embedding == null || params.embedding.length == 0) {\n          return null;\n      }\n      def shape = [params.embedding.length];\n      def json = \"{\" +\n                 \"\\\"name\\\":\\\"\" + name + \"\\\",\" +\n                 \"\\\"data_type\\\":\\\"\" + dataType + \"\\\",\" +\n                 \"\\\"shape\\\":\" + shape + \",\" +\n                 \"\\\"data\\\":\" + params.embedding +\n                 \"}\";\n      return json;\n    "
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Note that Bedrock access keys expire in 10--12 hours after creation. To learn more about the parameters in the preceding request, see [Connector blueprints]({{site.url}}{{site.baseurl}}/ml-commons-plugin/extensibility/blueprints#configuration-options).
```

Comment:
`10--12 hours after creation` where is this from? How about add some document link? 

I think this is for using [temporary credential](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_use-resources.html) case, right? 

In 2.11, we released a experimental API to allow user to update connector(see examples on this [PR](https://github.com/opensearch-project/ml-commons/pull/1484)) , so user can rotate the credential.